src/material/toolbar/toolbar.scss

```diff
@@ -56,6 +56,8 @@ $height-mobile-portrait: 56px !default;
     @include token-utils.use-tokens(
       tokens-mat-toolbar.$prefix, tokens-mat-toolbar.get-token-slots()) {
       $color-token: token-utils.get-token-variable(container-text-color);
+      // Update icon button color to match label text color for consistency while overriding.
+      --mdc-icon-button-icon-color: $color-token;
```

Comment:
We should do this through `use-tokens`, instead of hardcoding the name. See in the next few lines how it's done for the outlined and text buttons.